rclcpp/test/rclcpp/node_interfaces/test_node_services.cpp

```diff
@@ -103,7 +103,7 @@ TEST_F(TestNodeService, add_service_rcl_trigger_guard_condition_error)
     "lib:rclcpp", rcl_trigger_guard_condition, RCL_RET_ERROR);
   RCLCPP_EXPECT_THROW_EQ(
     node_services->add_service(service, callback_group),
-    std::runtime_error("Failed to notify wait set on service creation: error not set"));
```

Comment:
This seems like a step backwards... this is, perhaps, another reason to not introduce the `trigger_notify_guard_condition()` method?

Or it should have an error message argument to augment this error message. Or maybe all the callers should catch and rethrow. Either way, it would be better if these errors didn't change in this way.

Reply:
I'm not sure of the correct or best way to implement the augment of the error message. Should be passed as argument when calling `guard_condition->trigger("some error message")` ?

Regarding the introduction of `trigger_notify_guard_condition`, I can always remove it, so first we have to get the node's notify guard condition and then trigger it.

Reply:
I would say we should catch the throw from trigger and rethrow with more information.

Reply:
Done in 765257c